airflow/providers/amazon/aws/hooks/elasticache.py

```diff
@@ -0,0 +1,265 @@
+        self.initial_poke_interval = initial_poke_interval
+
+        super().__init__(client_type='elasticache', *args, **kwargs)
+
+    def create_replication_group(self, config):
+        """
+        Call ElastiCache API for creating a replication group
+
+        :param config: Python dictionary to use as config for creating replication group
+        :return: Response from ElastiCache create replication group API
+        """
+        return self.conn.create_replication_group(**config)
+
+    def delete_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for deleting a replication group
+
+        :param replication_group_id: ID of replication group to delete
+        :return: Response from ElastiCache delete replication group API
+        """
+        return self.conn.delete_replication_group(ReplicationGroupId=replication_group_id)
+
+    def describe_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for describing a replication group
+
+        :param replication_group_id: ID of replication group to describe
+        :return: Response from ElastiCache describe replication group API
+        """
+        return self.conn.describe_replication_groups(ReplicationGroupId=replication_group_id)
+
+    def get_replication_group_status(self, replication_group_id):
+        """
+        Get current status of replication group
+
+        :param replication_group_id: ID of replication group to check for status
+        :return: Current status of replication group
+        """
+        return self.describe_replication_group(replication_group_id)[self.REPLICATION_GROUPS][0][self.STATUS]
+
+    def is_replication_group_available(self, replication_group_id):
+        """
+        Helper for checking if replication group is available or not
+
+        :param replication_group_id: ID of replication group to check for availability
+        :return: True if available else False
+        """
+        return self.get_replication_group_status(replication_group_id) == self.STATUS_AVAILABLE
+
+    def _has_reached_terminal_state(self, replication_group_id):
+        """
+        Helper for checking if we should stop poking replication group for availability or not
+
+        :param replication_group_id: ID of replication group to check
+        :return: Flag to check if availability check should be stopped or not and current status
+        """
+        status = self.get_replication_group_status(replication_group_id=replication_group_id)
+
+        return status in (
+            self.STATUS_AVAILABLE,
+            self.STATUS_CREATE_FAILED,
+            self.STATUS_DELETING
+        ), status
+
+    def wait_for_availability(
+        self,
+        replication_group_id,
+        initial_sleep_time=None,
+        exponential_back_off_factor=None,
+        max_retries=None
+    ):
+        """
+        Check if replication is available or not by performing a describe over it
```

Comment:
This doc talks about replication -- but is it just health of elasticache? (Because this still applies when a single node EC is used, right?)